request.go

```diff
@@ -213,6 +213,22 @@ func (r Request) ResponseWithCode(body interface{}, statusCode int) Response {
 	return rsp
 }
 
+// RouterEndpointPattern finds the router pattern that matches the request. This is only callable while the request
+// is being served.
+func (r Request) RouterEndpointPattern() string {
```

Comment:
`RequestMethod` feels intuitive and simple. 👍 

For this method, [`RouterEntryPattern`](https://github.com/monzo/typhon/blob/ea17ae8b9d577dfbeeedf388993b22ba52867da5/router.go#L24) is probably the most correct, but `RequestPathPattern` or even `RequestPattern` would be simplest?